### PR TITLE
Add MNIST example with tensorboard

### PR DESCRIPTION
**Pull request by vfdev-5**
PR addresses #12.

Check with commands
```python mnist_with_tensorboard.py --log_dir=tensorboard_logs``` 
and 
```﻿python -m tensorboard.main --logdir=tensorboard_logs/```

Example writes summaries to plot : 
- Training loss
- Validation loss, accuracy
- if has `onnx` should plot model graph

### CODE REVIEW

**Comment by alykhantejani** [comment]
Thanks @vfdev-5 - I'll just verify by running locally - but then looks good to merge!

File: examples/mnist_with_tensorboard.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+import logging
+
+from torch.autograd import Variable
+from torch.utils.data import DataLoader
+from torch import nn
+import torch.nn.functional as F
+from torch.optim import SGD
+from torchvision.datasets import MNIST
+from torchvision.transforms import Compose, ToTensor, Normalize
+from tensorboardX import SummaryWriter
```
Comment (alykhantejani) [inline examples/mnist_with_tensorboard.py]: can we wrap this with a try except and raise an error telling the user they need to install the dependency

File: examples/mnist_with_tensorboard.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+                              batch_size=batch_size, shuffle=True)
+
+    val_loader = DataLoader(MNIST(download=False, root=".", transform=data_transform, train=False),
+                            batch_size=val_batch_size, shuffle=False)
+
+    model = Net()
+    optimizer = SGD(model.parameters(), lr=lr, momentum=momentum)
+
+    try:
+        writer.add_graph_onnx(model)
```
Comment (alykhantejani) [inline examples/mnist_with_tensorboard.py]: what does this do?

Comment (vfdev-5) [reply examples/mnist_with_tensorboard.py]: This should add a graph of the model using onnx:
See [this](https://github.com/lanpa/tensorboard-pytorch/blob/master/demo_graph.py) and [this](https://github.com/lanpa/tensorboard-pytorch/blob/master/tensorboardX/writer.py#L395)

**Comment by alykhantejani** [comment]
Thanks @vfdev-5!
